### PR TITLE
fixes #91: allow secure variant of scheme when only insecure is given

### LINKED ISSUE #91: allow secure variant of scheme when only insecure is given
See recent spec change here: https://github.com/w3c/webappsec-csp/issues/25#event-448827951

### PR DESCRIPTION
**Pull request by shekyan**

### CODE REVIEW

**Comment by michaelficarra** [comment]
Cool! Initial once-over looks good. I'll do a more detailed review soon.

File: src/main/java/com/shapesecurity/salvation/directiveValues/HostSource.java [hunk 1]
```diff
@@ -72,11 +72,18 @@ public boolean isWildcard() {
...
         boolean schemeMatches;
         if (this.scheme == null) {
-            schemeMatches = resource.scheme.equalsIgnoreCase("http") ?
-                shpOrigin.scheme.equalsIgnoreCase("http") || shpOrigin.scheme.equalsIgnoreCase("https") :
+            schemeMatches = resource.isNetworkScheme() ?
+                shpOrigin.isNetworkScheme() :
                 resource.scheme.equalsIgnoreCase(shpOrigin.scheme);
         } else {
-            schemeMatches = this.scheme.equalsIgnoreCase(resource.scheme);
+            if (resource.isNetworkScheme() && !resource.scheme.equalsIgnoreCase("ftp") && !this.scheme.equalsIgnoreCase("ftp")) {
```
Comment (michaelficarra) [inline src/main/java/com/shapesecurity/salvation/directiveValues/HostSource.java]: I can't find the reason for this `ftp` exception.

Comment (shekyan) [inline src/main/java/com/shapesecurity/salvation/directiveValues/HostSource.java]: We don't want to match `ftp` to any other network scheme.

**Comment by michaelficarra** [comment]
@sergey LGTM. But you either need to sign the CLA with your personal email address or amend your commit with your work email.

**Comment by shekyan** [comment]
@michaelficarra, that's strange. My both email addresses are in CLA for ages.

**Pull request merged**